client/packages/common/src/ui/forms/JsonForms/useJsonForms.tsx

```diff
@@ -80,28 +81,50 @@ const ScrollFix = () => {
   return null;
 };
 
+export type SavedDocument = {
+  id: string;
+  name: string;
+  type: string;
+};
+
 export type SaveJob = (
   jsonData: unknown,
   formSchemaId: string,
-  parent?: DocumentFragment
-) => Promise<void>;
+  parent?: string
+) => Promise<SavedDocument>;
 
 interface JsonFormOptions {
   showButtonPanel?: boolean;
   onCancel?: () => void;
   saveJob?: SaveJob;
+  onJobSaved?: (document: SavedDocument) => void;
   saveConfirmationMessage?: string;
   cancelConfirmationMessage?: string;
   saveSuccessMessage?: string;
 }
 
+/**
+ * Information required to create a new document
+ */
+export interface CreateDocument {
+  data: any;
+  documentRegistry: DocumentRegistryFragment;
+}
+
 export const useJsonForms = (
-  docName: string,
-  options: JsonFormOptions = {}
+  docName: string | undefined,
+  options: JsonFormOptions = {},
+  createDoc?: CreateDocument
 ) => {
-  const { data: databaseResponse, isLoading } =
-    useDocument.document.get(docName);
   const [data, setData] = useState<JsonData | undefined>();
+  // the current document id (undefined if its a new document)
+  const [documentId, setDocumentId] = useState<string | undefined>();
+  // document name can change from the input parameter when creating a new document
+  const [documentName, setDocumentName] = useState<string | undefined>();
+  useEffect(() => {
+    setDocumentName(docName);
+  }, []);
```

Comment:
```suggestion ts
   const [documentName, setDocumentName] = useState<string | undefined>(docName);
```

You should be able to get rid of this useEffect if you specify a initial value for the state variable